content/general/push/activate-subscribe.textile

```diff
@@ -373,3 +373,57 @@ blang[android].
       }
   }
   ```
+
+h1. Push API reference
+
+inline-toc.
+  Methods:
+    - activate()#activate
+    - deactivate()#deactivate
+  Related types:
+    - DeviceDetails#device-details
+    - LocalDevice#localdevice
+
+h2(#push-object). Push object
+
+This object is accessible through @client.push@ and provides to "push-compatible devices":/general/push#platform-support:
+
+h3. Methods
+
+h6(#activate).
+  default: activate
+
+bq(definition).
+  android:    void activate()
+  objc,swift: activate(callback: ("ARTErrorInfo":/realtime/types#error-info?, DeviceDetails?) -> Void)
+
+Register the device for push. When the "activation process":#device-activation is completed, Ably will <span lang="android">send a broadcast through the application's "@LocalBroadcastManager@":https://developer.android.com/reference/android/support/v4/content/LocalBroadcastManager. Success or failure will be broadcast through @io.ably.broadcast.PUSH_ACTIVATE@</span><span lang="objc,swift">call the </span><span lang="objc">@(void)didActivateAblyPush:(nullable ARTErrorInfo *)error@</span><span lang="swift">@didActivateAblyPush(error: ARTErrorInfo?)@</span><span lang="objc,swift"> method from the @ARTPushRegistererDelegate@</span>.
+
+h6(#deactivate).
+  default: deactivate
+
+bq(definition).
+  android:    void deactivate()
+  objc,swift: deactivate(deregisterCallback: ("ARTErrorInfo":/realtime/types#error-info?, deviceId: String?) -> Void)
+
+Deregister the device for push. When the deactivation process is completed, Ably will <span lang="android">send a broadcast through the application's "@LocalBroadcastManager@":https://developer.android.com/reference/android/support/v4/content/LocalBroadcastManager. Success or failure will be broadcast through @io.ably.broadcast.PUSH_DEACTIVATE@</span><span lang="objc,swift">call the </span><span lang="objc">@(void)didDeactivateAblyPush:(nullable ARTErrorInfo *)error@</span><span lang="swift">@didDeactivateAblyPush(error: ARTErrorInfo?)@</span><span lang="objc,swift"> method from the@ARTPushRegistererDelegate@</span>.
+
+h2(#related-types). Related types
+
+h3(#device-details).
+  default: DeviceDetails
+  ruby:    Ably::Models::DeviceDetails
+
+<%= partial 'types/_device_details' %>
+
+h3(#local-device).
+  default: LocalDevice
+  ruby:    Ably::Models::LocalDevice
+
+<%= partial 'types/_local_device' %>
+
+h3(#push-channel).
+    default: PushChannel
+    ruby:    Ably::Models::PushChannel
+
+<%= partial 'types/_push_channel' %>
```

Comment:
I assume you'll need to update these all to use `partial_version` once all the work is merged.